### PR TITLE
MCC-206116,MCC-222759 : Adding self-user registration and uploading data to S3 bucket.

### PR DESCRIPTION
**Pull request by ssucharitdas**
PR includes:
- User can use the sample to self register
- Once logged in, user can use the Capture Image form to take an image or load one from library and upload to AWS S3.

### CODE REVIEW

File: AppConnectSwift/Babbage.podspec [hunk 1]
```diff
@@ -3,17 +3,17 @@ password = ENV['ARTIFACTORY_PASSWORD'] || raise("You must set an artifactory pas
...
-    s.source             = { http: "https://#{username}:#{password}@etlhydra-artifactory-sandbox.imedidata.net/artifactory/p-cloud-release/com/mdsol/babbage/ios/2016.1.0/babbage-2016.1.0.18.zip" }
+    s.source             = { http: "https://#{username}:#{password}@etlhydra-artifactory-sandbox.imedidata.net/artifactory/p-cloud-release/com/mdsol/babbage/ios/2016.2.0/babbage-2016.2.0.81.zip" }
     s.source_files       = "artifacts/include/babbage/*.h"
     s.vendored_libraries = "artifacts/libBabbage.a"
 
-    s.library            = "c++"
+    s.library            = 'c++', 'z'
     s.platform           = :ios
     s.requires_arc       = true
     s.ios.deployment_target = '7.0'
```
Comment (marrodriguez) [inline AppConnectSwift/Babbage.podspec]: do we need to keep this at 7.0?

Comment (kbohlmann-mdsol) [inline AppConnectSwift/Babbage.podspec]: Good question - we are releasing Epro with minimum of 8.0, right?

File: AppConnectSwift/AppDelegate.swift [hunk 2]
```diff
@@ -16,7 +16,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // long and must be the same between runs.
         let dir = NSFileManager.defaultManager().URLsForDirectory(NSSearchPathDirectory.DocumentDirectory, inDomains: NSSearchPathDomainMask.UserDomainMask).last
         let key = "12345678901234567890123456789012".dataUsingEncoding(NSUTF8StringEncoding)
-        MDBabbage.startWithDatastoreAtURL(dir, encryptionKey: key)
+        // TODO: Substitute the apiToken value with proper token
+        MDBabbage.startWithEnvironment(MDClientEnvironment.Sandbox, apiToken: "Your API token", publicDirectory: dir, privateDirectory: dir, encryptionKey: key)
```
Comment (marrodriguez) [inline AppConnectSwift/AppDelegate.swift:21]: are we keeping the environment Sandbox?

**Comment by kbohlmann-mdsol** [comment]
Looks like we are ready to go. 🐩

**Pull request merged**